src/org/zaproxy/zap/extension/invoke/InvokeAppWorker.java

```diff
@@ -101,7 +109,19 @@ protected Void doInBackground() throws Exception {
 						.replace("%port%", port)
 						.replace("%site%", site)
 						.replace("%cookie%", cookie)
-						.replace("%postdata%", postdata);
+						.replace("%postdata%", postdata)
+						.replace("%msgid%", String.valueOf(msgid));
+
+				// Replace header tags
+				Matcher headers = Pattern.compile("%header-([A-z0-9_-]+)%").matcher(finalParameter);
+				while (headers.find()) {
+					String headerValue = msg.getRequestHeader().getHeader(headers.group(1));
+					if (headerValue == null) {
+						headerValue = "";
+					}
+					finalParameter = finalParameter.replace("%header-" + headers.group(1) + "%", headerValue);
```

Comment:
Still pushing for `headers.group(0)`, less token duplication to maintain (if the usage is not clear it can be assigned to a variable).

Reply:
Ah apologies, I understand now what you mean, I did not compute fully the first time you mentioned that!